From: Karel Zak <kzak@redhat.com>
Date: Mon, 11 May 2009 09:49:00 +0000 (+0200)
Subject: build-sys: fix "make -C" bug
X-Git-Url: http://git.kernel.org/?p=utils%2Futil-linux-ng%2Futil-linux-ng.git;a=commitdiff_plain;h=2ef7d41e6bbb7c842c06e911ffc562335a467b5d

build-sys: fix "make -C" bug

Running:

./configure --with-fsprobe=builtin
make -C mount mount

causes the following error:

gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I..  -include ../config.h
-I../include -DLOCALEDIR=\"/home/matthew/util-linux-ng/share/locale\"
-fsigned-char -I../libs/blkid/src   -g -O2 -MT mount-setproctitle.o
-MD -MP -MF .deps/mount-setproctitle.Tpo -c -o mount-setproctitle.o
`test -f '../lib/setproctitle.c' || echo './'`../lib/setproctitle.c mv
-f .deps/mount-setproctitle.Tpo .deps/mount-setproctitle.Po
make -C
make: option requires an argument -- 'C'
Usage: make [options] [target] ...
...

Reported-by: Matthew Burgess <matthew@linuxfromscratch.org>
Signed-off-by: Karel Zak <kzak@redhat.com>
---

diff --git a/config/include-Makefile.am b/config/include-Makefile.am
index b42a0c3..d5da619 100644
--- a/config/include-Makefile.am
+++ b/config/include-Makefile.am
@@ -17,6 +17,7 @@ dist_noinst_DATA = $(dist_man_MANS)
 # Paths to in-tree libraries (use ul_ prefix to avoid possible collisions)
 #
 ul_libblkid_srcdir = $(top_srcdir)/libs/blkid/src
+ul_libblkid_builddir = $(top_builddir)/libs/blkid/src
 ul_libblkid_la = $(top_builddir)/libs/blkid/src/libblkid.la
 
 $(ul_libblkid_la):
